### ISSUE TITLE
(Suggestions/Ideas) Different suggestions for playlists

### ISSUE THREAD

**Issue by imkrut** [libretro/RetroArch#3038]
Just ideas/suggestions/things to consider down the line, thanks to everyone working on this fantastic project. Please excuse me if this isn't the best way to voice them or the best place to do so.

Suggestion 1: 

Add a "played" (y/n) or "number of times played" counter to each game on the playlist.
It's specially useful for trying out new games in full-romsets or keeping track of which particular rom is being played.

Suggestion 2:  
Make playlists for a particular system able to be modified on the fly by hiding certain games/ or marking some games as "favourites" on that particular playlist and hiding the rest.
The optional ability to "refresh" (automatically adding or removing changed games) or the ability to manually add games to a playlist from within the client while browsing files. (similar to manually creating custom playlists as noted here: https://github.com/libretro/RetroArch/wiki/Manually-Creating-Custom-Playlists )

Suggestion 2.5:  
Ability to set certain paramaters (based on the No-intro guidelines if wanted) to hide duplicate games/hide certain regions. Also if implemented it would be nice to have to option to modify on the fly the title shown hiding the region/languages and just show the title of the game.
For example instead of showing "4x4 World Trophy (Europe) (En,Fr,De,Es,It)"
have to option to toggle "4x4 World Trophy (Europe)" or "4x4 World Trophy"

Suggestion 3: 
Ability to group/collapse playlist, for example, gameboy (mono) and color grouped as one, and expand it if needed/wanted.

Suggestion 4: 
Set a X/Y size frame for the thumbnails and stretch the corresponding image to that particular size.
Yes, obviously can be done manually, but it would be nice to have the option as a way to keep a standard on the images shown.

Suggestion 5:
Ability to use a placeholder image reprensenting of the system for all the games that don't have that particular image file available.

**Comment by Kivutar** [comment]
1) It can be done, by changing the playlist format. I don't think it would be difficult. It could be used to generate a list of most played games too. It would break compatibility with the existing playlists and users would have to regenerate them.

2) @fr500 had some proposal about it too. I'm against adding more entries to the list itself. I would like to implement a side menu like this for all this kind of actions: http://psx-scene.com/forums/content/attachments/29996d1314357617-sidesettingsx.jpg/

2.5) I was thinking about that too. Easy to implement I think. We could also move everything in parenthesis to a smaller label under the title.

3) Too difficult

4) I'm not sure to understand. How can this be done manually?

5) Ok, not too difficult. But it will become useless once the thumbnails are completed.

**Comment by imkrut** [comment]
Thanks for addressing my suggestions Kivutar, nice to see I posted it correctly/in the right place.

Regarding : 

4) Let me rephrase.

The problem currently is different sizes for covers/boxarts/screenshots (title/snapshots) between systems (annoying when browsing different systems), and even in a single system there are discrepancies between screen size (for example, sega genesis, there are some games with different resolutions iirc).

Solution currently: Manually I batch resize them all to a standard size (using an external program: faststone photo resizer) 

The problem with that solution is that it alters the actual 1:1 ratio permanently of the image used.

5) Is not entirely useless even when thumbnails are completed, because for example there are roms that didn't have a physical release/box release, and you could use the  placeholder for that purpose, not to mention there are tons of images not currently available on packs.


Another interesting idea and somewhat related with another topic, is since per-game options are now a thing (yay!) the ability to select and "attach" a certain IPS to a rom. This would allow for an easier launch of translated roms using the playlist or hacks, giving the best of two worlds (Goodsets and no-intro). Because I don't know if it's just me, but I'm unable to launch soft patched (Translated) snes roms (using any core, identical file name except extension) I get a black screen, however if I hard patch it, it runs flawlessly (but obviously doesn't show up on the playlist anymore unless manually added).

This would solve this, and open up the possibility of launching for example multiple game hacks, from within the client without cluttering a gamelist, or even giving the option to launch MSU-1 enhancement for what is basically is the same game.

Another fun idea (Again just speaking out of my ass, since I have no idea how hard it is to implement this) would be to have dynamic wallpapers associated to the game currently selected. So for example if I'm currently playing/or highlighting "Donkey Kong Country" on the SNES, then the background changes to snapshots that i've taken of that game (for this purposes, it would be swell if screenshots taken would adopt the filename of the rom currently being played instead of the generic "Retroarchxxxx") In fact that would also help people that want to take their own screenshots/screencaps for the title or whatever (right now there's no way to identify them).

Thanks for reading

**Comment by Kivutar** [comment]
4) The current thumbnails all have the same width (except for famous games, where we use a double resolution). Then RetroArch will display them at a fixed width, always at the same place. The only thing that varies is the height of the image. I don't get yet how it is different from your proposal.

5) Interesting!

**Comment by imkrut** [comment]
4) Oh, wasn't aware the width was fixed already! I meant giving the option to also "fix" the height. That way there's a consistency on size (and images don't jump around all over the place).

I'm attaching an example of what i mean:

![auto-resize](https://cloud.githubusercontent.com/assets/10028924/15619986/ef5ebe1e-242d-11e6-82fd-e6f64a776fc2.png)

Top and bottom are current SNES and VBoy images, I propose image on the right, in which the client resizes the image "on the fly" to a certain width and height (that way the original file keeps their original ratio and sizes), same could be said about boxarts.

The reason why I think custom width is also important is because there are systems that due to regional  releases have different orientation/sizes. For example SNES and Super Famicom

**Comment by ofry** [comment]
Next suggestion:

make recursive subfolder scanning option in playlist generation. Could you do that?

**Comment by ofry** [comment]
Next suggestion:

Rescan playlist.
This feature would scan existing items from this playlist only! And it would give them actual info from databases.

Why this feature need? Because databases contain some errors. So, if some errors in referensed databases will fix, I want to rescan my playlist to fix these errors in playlist data.

**Comment by RobLoach** [comment]
- Suggestion 1: Last played sublabel is in there now
- Suggestion 2: Favorites are in there now
- Suggestion 2.5: You can hide certain sections of labels now
- Rescan playlist... You can scan a directory, it shouldn't add duplicates.

If you still want the other suggestions, feel free to create specific issues for each to discuss. Issues can get confusing when tackling many at a time. Thanks!
